From a282dd8f5f82adee27363c65c843ba1b2baf6035 Mon Sep 17 00:00:00 2001
From: Paul Gofman <pgofman@codeweavers.com>
Date: Fri, 24 Jun 2022 18:53:58 -0500
Subject: [PATCH 0274/1306] winex11.drv: Add an option to always Increment
 configure_serial in X11DRV_WindowPosChanged().

And enable it for Dokyusei Bangin' Summer.

CW-Bug-Id: #20665
---
 dlls/winex11.drv/window.c | 24 +++++++++++++++++++++---
 1 file changed, 21 insertions(+), 3 deletions(-)

diff --git a/dlls/winex11.drv/window.c b/dlls/winex11.drv/window.c
index b90074ae65d..08e88ac4ddd 100644
--- a/dlls/winex11.drv/window.c
+++ b/dlls/winex11.drv/window.c
@@ -3052,6 +3052,20 @@ done:
     return TRUE;
 }
 
+static BOOL option_increament_configure_serial(void)
+{
+    static int increment = -1;
+    if (increment == -1)
+    {
+        const char *e = getenv( "WINE_INCREMENT_CONFIGURE_SERIAL" );
+
+        if (e)
+            increment = atoi( e );
+        else
+            increment = (e = getenv( "SteamGameId" )) && !strcmp( e, "1689910" );
+    }
+    return increment;
+}
 
 static void restack_windows( struct x11drv_win_data *data, HWND prev )
 {
@@ -3188,9 +3202,13 @@ void X11DRV_WindowPosChanged( HWND hwnd, HWND insert_after, UINT swp_flags,
     }
 
     /* don't change position if we are about to minimize or maximize a managed window */
-    if ((!event_type || event_type == PropertyNotify) &&
-        !(data->managed && (swp_flags & SWP_STATECHANGED) && (new_style & (WS_MINIMIZE|WS_MAXIMIZE))))
-        prev_window = sync_window_position( data, swp_flags, &old_window_rect, &old_whole_rect, &old_client_rect );
+    if (!event_type || event_type == PropertyNotify)
+    {
+        if (!(data->managed && (swp_flags & SWP_STATECHANGED) && (new_style & (WS_MINIMIZE|WS_MAXIMIZE))))
+            prev_window = sync_window_position( data, swp_flags, &old_window_rect, &old_whole_rect, &old_client_rect );
+        else if (option_increament_configure_serial())
+            data->configure_serial = NextRequest( data->display );
+    }
 
     if ((new_style & WS_VISIBLE) &&
         ((new_style & WS_MINIMIZE) || is_window_rect_mapped( rectWindow )))
-- 
2.43.2

